daemon/qrexec-client.c

```diff
@@ -70,7 +76,9 @@ static int handle_agent_handshake(libvchan_t *vchan, int remote_send_first)
                 return -1;
             }
 
-            if (info.version != QREXEC_PROTOCOL_VERSION) {
+            data_protocol_version = info.version < QREXEC_PROTOCOL_VERSION ?
+                                    info.version : QREXEC_PROTOCOL_VERSION;
+            if (data_protocol_version < 2) {
```

Comment:
Formally hardcoded constants are against the style guide. I personally think this is fine but maybe consider to use something like `QREXEC_MIN_VERSION` or so.

There are a few other cases of hardcoded 2s and 3s. I won't tag them all.